nequip/train/trainer.py

```diff
@@ -883,13 +884,13 @@ def epoch_step(self):
                     )
                     self.end_of_batch_log(batch_type=category)
                     for callback in self.end_of_batch_callbacks:
-                        callback(self)
+                        load_callable(callback)(self)
```

Comment:
can't we `self.end_of_batch_callbacks = [load_callable(c) for c in end_of_batch_callbacks]` in `__init__` instead?
seems wasteful to call down into YAML every single epoch

similar point for the other callback lists

Reply:
yes. I need to rename the variable though. otherwise the dict saving will save the functions instead of the arguments.

Reply:
aha
one easier option maybe is just to memoize the load_callback function?
have it keep a dict so it doesn't call to load YAML repeatedly @nw13slx

Reply:
sorry. my previous message was prefixed. I have revised it as you suggested.

Reply:
just use something like "self._end_of_xxx_callbacks" to store the functions

Reply:
great